modules/openid_auth/app/controllers/openid_auth/v2/validation_controller.rb

```diff
@@ -81,6 +92,24 @@ def setup_structure
         end
         payload_object
       end
+
+      def charon_token_screen?(payload_object)
+        vid = payload_object.act[:vista_id]
+        sta3n = payload_object.launch['sta3n']
+        return false unless !vid.nil? && !sta3n.nil?
+
+        parsed_sta3n = vid.match(/\d{3}[A-Z]*\|\d+\^[A-Z]{2}\^\d{3}[A-Z]*\^[A-Z]{5}\|[A-Z]{1}/)
+        return false unless parsed_sta3n
+
+        parsed_sta3n = parsed_sta3n[0].match(/\d{3}|/)
```

Comment:
We're told that _most_ users will only have a single vista_id, but it's possible for there to be more than one (as with the test user). I think we need to check them all.

The direction in the story (and other conversations) may have not implied this, and it was probably my bad.

> Ensure that `act.vista_id` contains an entry for `launch.sta3n`

Reply:
It was clear and forgot to finish a redo. My original implementation covered this but it was much looser check.